pkg/run/run.go

```diff
@@ -189,12 +189,15 @@ func (r *Runtime) Run() error {
 				continue
 			}
 			// kubectl apply -f manifests-api
-			err = r.applyManifests()
-			if err != nil {
-				// TODO do we trigger an exit at some point
-				// TODO because it's almost a deadlock if the user didn't set a short --timeoutTimer
-				glog.Errorf("Cannot apply manifests in %s", r.env.GetManifestsPathToApply())
-				continue
+			if !r.state.IsKubectlApplied() {
+				err = r.applyManifests()
+				if err != nil {
+					// TODO do we trigger an exit at some point
+					// TODO because it's almost a deadlock if the user didn't set a short --timeoutTimer
+					glog.Errorf("Cannot apply manifests in %s", r.env.GetManifestsPathToApply())
+					continue
+				}
+				r.state.SetKubectlApplied()
```

Comment:
Why don't we run a r.UnsetKubectlApplied()
```
// SetKubectlApplied mark the state when kubectl apply successfully returned
func (s *State) UnsetKubectlApplied() {
	s.Lock()
	s.kubectlApplied = false
	s.Unlock()
}
```
around: https://github.com/DataDog/pupernetes/blob/master/pkg/run/delete.go#L322 ?

Reply:
We would need more code about the namespace no?

Reply:
yes, for sure - I just want to make sure resetting allows for a fresh start and thus one could reset the state 2, 3, N times in a row with no issues.

Reply:
The reset capability is managed by a dedicated channel, so I think it's safe and simple to let it like that and this restore the previous logic.
Then, it's not that bad if we apply when not needed because the kubectl apply is idempotent, this is safe.
Today we only have some resources in the kube-system namespace, but we could have something else in an another namespace. It let to the user a way to configure it as it suits him.